test/test_torch.py

```diff
@@ -1126,18 +1126,25 @@ def test_pdist_empty(self):
             self.assertEqual(torch.zeros(3, device=device), torch.pdist(x))
 
     def test_pdist_norm(self):
+        def test_pdist_single(shape, device, p, trans):
+            x = torch.randn(shape, device=device)
+            if trans:
+                x.transpose_(-2, -1)
+            actual = torch.pdist(x, p=p)
+            expected = brute_pdist(x, p=p)
+            self.assertEqual(expected.shape, actual.shape)
+            self.assertTrue(torch.allclose(expected, actual))
+
         devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
         for device in devices:
-            for shape in [(4, 5), (3, 2), (2, 1), (2, 3, 4)]:
+            for shape in [(4, 5), (3, 2), (2, 1)]:
                 for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
                     for trans in [False, True]:
-                        x = torch.randn(shape, device=device)
-                        if trans:
-                            x.transpose_(-2, -1)
-                        actual = torch.pdist(x, p=p)
-                        expected = brute_pdist(x, p=p)
-                        self.assertEqual(expected.shape, actual.shape)
-                        self.assertTrue(torch.allclose(expected, actual))
+                        test_pdist_single(shape, device, p, trans)
+
+            # do a simplified comparison with big inputs, see:
+            # https://github.com/pytorch/pytorch/issues/15511
+            test_pdist_single((1000, 2), device, 2, False)
```

Comment:
15511 failed for a size of 10000 due to indices computation, not 1000. Not sure though how feasible it is to actually run correctness tests for 10000 though. Also, can we add pdist backward test? I suspect it will fail for smallish sizes because of https://github.com/pytorch/pytorch/blob/2134d8f0838e99794c4fece8599465c377d313e0/aten/src/ATen/native/cuda/DistanceKernel.cu#L193

Reply:
I ran the script in the issue and it failed for me at 1000.  I tried 10000 and the test is valid, it just takes much longer (on my machine, ~11 seconds instead of ~1.5 seconds).  I'm actually not sure if that matters or we run the tests in parallel anyway -- @yf225?

Reply:
and yes, I'll add a backwards test.

When do you expect that `grid_y` line to cause a problem?  Is it when `grid_y` is > 2^16, so `numel` of around 2^18 (262144), because `block_y` is 4?

Reply:
yes, numel of 2^18, but since `numel = n*(n+1)/2` if I'm not mistaken, that corresponds to n of about 720.